packages/compiler-dom/__tests__/transforms/vCloak.spec.ts

```diff
@@ -0,0 +1,36 @@
+import {
+  parse,
+  transform,
+  CompilerOptions,
+  ElementNode,
+  NodeTypes
+} from '@vue/compiler-core'
+import { transformCloak } from '../../src/transforms/vCloak'
+
+function transformWithCloak(template: string, options: CompilerOptions = {}) {
+  const ast = parse(template)
+  transform(ast, {
+    directiveTransforms: {
+      cloak: transformCloak
+    },
+    ...options
+  })
+  return {
+    root: ast,
+    node: ast.children[0] as ElementNode
+  }
+}
+
+describe('compiler: `v-cloak` transform', () => {
+  test('should add no props to DOM', () => {
+    const { node } = transformWithCloak(`<div v-cloak/>`)
+
+    expect(node.props.length).toBe(1)
```

Comment:
We should be asserting `node.codegenNode` similar to [this](https://github.com/vuejs/vue-next/blob/master/packages/compiler-core/__tests__/transforms/vBind.spec.ts#L38) (where we'd expect a properties length of 0)